packages/terra-application-layout/src/ApplicationLayout.jsx

```diff
@@ -0,0 +1,222 @@
+   * the ApplicationLayout only supports configuration for the `menu` and `content` regions of the layout. The '/' path is also blacklisted
+   * within this configuration object, as the ApplicationLayout uses it for navigation purposes. Any configuration provided for the '/' path
+   * will be disregarded.
+   */
+  routingConfig: PropTypes.shape({
+    menu: routeConfigPropType,
+    content: routeConfigPropType,
+  }),
+  /**
+   * The content to be rendered in the ApplicationLayout's extensions region.
+   */
+  extensions: PropTypes.element,
+};
+
+const defaultProps = {
+  navigationItems: [],
+};
+
+class ApplicationLayout extends React.Component {
+  /**
+   * Builds and returns the menu items for the PrimaryNavigationSideMenu from the navigationItems.
+   */
+  static buildMenuNavigationItems(props) {
+    const { navigationItems, routingConfig } = props;
+
+    if (!routingConfig.menu) {
+      return navigationItems;
+    }
+
+    const menuPaths = Object.keys(routingConfig.menu).map(key => (routingConfig.menu[key].path));
+    return navigationItems.map(navigationItem => ({
+      path: navigationItem.path,
+      text: navigationItem.text,
+      hasSubMenu: menuPaths.filter(menuPath => matchPath(navigationItem.path, { path: menuPath })).length > 0,
+    }));
+  }
+
+  /**
+   * Builds and returns the routing configuration object for the PrimaryNavigationSideMenu.
+   */
+  static buildNavigationMenuConfig(props) {
+    const menuNavigationItems = ApplicationLayout.buildMenuNavigationItems(props);
+
+    const componentConfig = {
+      componentClass: PrimaryNavigationSideMenu,
+      props: {
+        navigationItems: menuNavigationItems,
+      },
+      refuseRoutingStackNavigation: menuNavigationItems.length === 0,
+    };
+
+    /**
+     * The configuration for the primary navigation menu is specified for the
+     * tiny and small breakpoints only. The menu will only be visible when the ApplicationLayout
+     * is compact.
+     */
+    return {
+      '/': {
+        path: '/',
+        component: {
+          tiny: componentConfig,
+          small: componentConfig,
+        },
+      },
+    };
+  }
+
+  /**
+   * Builds and returns the routing configuration object for all menus with ApplicationMenuConfigAdapters
+   * wrapped around each component entry.
+   */
+  static buildWrappedMenuConfig(props, menuConfig) {
+    const { nameConfig, utilityConfig, extensions } = props;
+
+    const updatedMenuConfig = {};
+    Object.keys(menuConfig).forEach((menuKey) => {
+      const newMenu = Object.assign({}, menuConfig[menuKey]);
+
+      const newMenuComponent = Object.assign({}, newMenu.component);
+
+      navigationLayoutSizes.forEach((size) => {
+        if (!newMenuComponent[size]) {
+          return;
+        }
+
+        const configForSize = Object.assign({}, newMenuComponent[size]);
+
+        const propsForSize = Object.assign({}, configForSize.props);
+
+        propsForSize.applicationMenuConfigAdapterProps = {
```

Comment:
Add comment like `inject ApplicationMenuConfigAdapter props to correctly render the supplied Menu config`